### PR TITLE
wire print flags through get.go

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Adds the PrintFlags pattern to `get.go`.
Prerequisite to ongoing server-side printing work.

cc @soltysh @deads2k

### CODE REVIEW

File: pkg/kubectl/cmd/get/humanreadable_flags.go [hunk 1]
```diff
@@ -14,13 +14,15 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package printers
+package get
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/get/humanreadable_flags.go]: @soltysh I moved these files to their own package under `cmd`, as I needed to break an import cycle when depending on `printersinternal.AddHandlers` below

File: pkg/kubectl/cmd/get/get.go [hunk 2]
```diff
@@ -49,8 +49,13 @@ import (
 	"k8s.io/kubernetes/pkg/util/interrupt"
 )
 
-// GetOptions contains the input to the get command.
-type GetOptions struct {
+// CmdOptions contains the input to the get command.
+type CmdOptions struct {
```
Comment (soltysh) [inline pkg/kubectl/cmd/get/get.go]: Why did you change to `CmdOptions`?

File: pkg/kubectl/cmd/get/get.go [hunk 3]
```diff
@@ -201,14 +206,49 @@ func (options *GetOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args
...
+	options.PrintGeneric = printer.IsGeneric()
+
 	options.IncludeUninitialized = cmdutil.ShouldIncludeUninitialized(cmd, false)
+	options.PrintWithOpenAPICols = cmdutil.GetFlagBool(cmd, useOpenAPIPrintColumnFlagLabel)
+
+	options.ToPrinter = func(mapping *meta.RESTMapping, withNamespace bool) (printers.ResourcePrinterFunc, error) {
+		// make a new copy of current flags / opts before mutating
+		printFlags := options.PrintFlags.Copy()
+
+		if mapping != nil && !cmdSpecifiesOutputFmt(cmd) && options.PrintWithOpenAPICols {
```
Comment (soltysh) [inline pkg/kubectl/cmd/get/get.go]:
```go
if mapping != nil {
    if !cmdSpecifiesOutputFmt(cmd) && options.PrintWithOpenAPICols {
        if apiSchema, err := f.OpenAPISchema(); err == nil {
            printFlags.UseOpenAPIColumns(apiSchema, mapping)
        }
        if resource.MultipleTypesRequested(args) {
            printFlags.EnsureWithKind(mapping.GroupVersionKind.GroupKind())
        }
}
```

File: pkg/kubectl/cmd/get/get.go [hunk 4]
```diff
@@ -165,18 +173,15 @@ func NewCmdGet(f cmdutil.Factory, out io.Writer, errOut io.Writer) *cobra.Comman
...
-	cmd.Flags().BoolVar(&options.ShowKind, "show-kind", options.ShowKind, "If present, list the resource type for the requested object(s).")
-	cmd.Flags().StringSliceVarP(&options.LabelColumns, "label-columns", "L", options.LabelColumns, "Accepts a comma separated list of labels that are going to be presented as columns. Names are case-sensitive. You can also use multiple flag options like -L label1 -L label2...")
 	cmd.Flags().BoolVar(&options.Export, "export", options.Export, "If true, use 'export' for the resources.  Exported resources are stripped of cluster-specific information.")
 	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, "identifying the resource to get from a server.")
 	return cmd
 }
 
 // Complete takes the command arguments and factory and infers any remaining options.
-func (options *GetOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+func (options *CmdOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
```
Comment (soltysh) [inline pkg/kubectl/cmd/get/get.go]: s/options/o 

see David's other PR.

File: pkg/kubectl/cmd/get/get.go [hunk 5]
```diff
@@ -299,8 +333,8 @@ func (options *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []str
 		return err
 	}
 
-	if printer.IsGeneric() {
-		return options.printGeneric(printer, r)
+	if options.PrintGeneric {
+		return options.printGeneric(r)
```
Comment (soltysh) [inline pkg/kubectl/cmd/get/get.go]: The `printGeneric` method is not about just printing, let's rename it to `getGeneric` or alike. The word `print` in it is confusing with the printers. wdyt?

File: pkg/printers/printers.go [hunk 6]
```diff
@@ -75,36 +75,34 @@ func GetStandardPrinter(typer runtime.ObjectTyper, encoder runtime.Encoder, deco
...
-		humanPrinter, matches, err := humanPrintFlags.ToPrinter(format)
-		if !matches {
-			return nil, fmt.Errorf("unable to match a printer to handle current print options")
-		}
-		if err != nil {
-			return nil, err
-		}
-
-		printer = humanPrinter
+	//case "wide":
```
Comment (soltysh) [inline pkg/printers/printers.go]: Remove?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/kubectl/cmd/get/humanreadable_flags.go (hunk 1 above)
Comment (soltysh) [reply pkg/kubectl/cmd/get/humanreadable_flags.go]: I wonder if we can get rid of it...

File: pkg/printers/printers.go (hunk 6 above)
Comment (juanvallejo) [reply pkg/printers/printers.go]: Yeah, left in there for review in case I had overlooked something that might break by removing this case

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
You need to rebase though :/

**Review by soltysh — APPROVED** [review]
/lgtm
